apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -362,25 +365,40 @@ export const getLocalisedSitemap = async (
         .leftJoin("Blob as published", "Version.blobId", "published.id")
         .select(() => [headerSql, thumbnailSql, ...defaultResourceSelect]),
     )
-    .with("childCousinIndexPages", (eb) => {
-      return eb
+    // Step 3: Get all nested folders and collections
+    .withRecursive("nestedResources", (eb) =>
```

Comment:
Actually can explain a bit more on why do we need this? My understanding was that we were injecting dummy data into the localisedSitemap we have fetched to render the collection block with 3 placeholder items

Reply:
@harishv7 here https://github.com/opengovsg/isomer/pull/1291#discussion_r2113582834 !

You are right that the children is using the dummy blocks, but the title and description aren't - they are using the values from the Collection (or Index Page, if any)

Reply:
that's a possible different approach, though personally I thought that was a bit less straightforward and different from how things are currently done (via the same single method), as opposed to this approach of just only tweaking the query (which is easier to test for regression via unit test too). Esp. since in most cases, the number of nestings wouldn't be as deep as this test case.

Reply:
Discussed offline, understand we want to show the users some preview of "actual" content when they use a collection block as have fully dummy data will not be ideal. Given the below benchmarks I think it's reasonable for the use case as well. 

Will go ahead to approve the PR